pkg/antctl/raw/upgrade/apistorage/command.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apistorage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	apiextinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"antrea.io/antrea/pkg/antctl/raw"
+)
+
+var example = strings.Trim(`
+  Upgrade all existing objects of Antrea CRDs to the latest API version
+  $ antctl upgrade api-storage
+
+  Upgrade existing AntreaAgentInfo objects to the latest API version
+  $ antctl upgrade api-storage --crds=antreaagentinfos.crd.antrea.io
+
+  Upgrade existing Egress and Group objects to the latest API version
+  $ antctl upgrade api-storage --crds=egresses.crd.antrea.io,groups.crd.antrea.io
+`, "\n")
+
+type options struct {
+	k8sClient client.Client
+	crdNames  []string
+}
+
+var opts *options
+
+func NewCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "api-storage",
+		Short:   "Upgrade existing objects of Antrea CRDs to the latest API version",
+		Example: example,
+		RunE:    runE,
+	}
+	o := &options{}
+	command.Flags().StringSliceVar(&o.crdNames, "crds", nil, "")
```

Comment:
Perhaps it would be more friendly to provide a "--dry-run" option, in which mode the command only output which CRD's objects will be upgraded to which version but not really upgrade them. Users can use this mode to understand what will happen.

Reply:
Added.